ev3dev2/display.py

```diff
@@ -23,420 +23,519 @@
-    # Copyright (c) 2012 Kurichan
-    #
-    # This program is free software. It comes without any warranty, to
-    # the extent permitted by applicable law. You can redistribute it
-    # and/or modify it under the terms of the Do What The Fuck You Want
-    # To Public License, Version 2, as published by Sam Hocevar. See
-    # http://sam.zoy.org/wtfpl/COPYING for more details.
-    # ------------------------------------------------------------------
-
-    __slots__ = ('fid', 'fix_info', 'var_info', 'mmap')
-
-    FBIOGET_VSCREENINFO = 0x4600
-    FBIOGET_FSCREENINFO = 0x4602
-
-    FB_VISUAL_MONO01 = 0
-    FB_VISUAL_MONO10 = 1
-
-    class FixScreenInfo(ctypes.Structure):
-
-        """The fb_fix_screeninfo from fb.h."""
-
-        _fields_ = [
-            ('id_name', ctypes.c_char * 16),
-            ('smem_start', ctypes.c_ulong),
-            ('smem_len', ctypes.c_uint32),
-            ('type', ctypes.c_uint32),
-            ('type_aux', ctypes.c_uint32),
-            ('visual', ctypes.c_uint32),
-            ('xpanstep', ctypes.c_uint16),
-            ('ypanstep', ctypes.c_uint16),
-            ('ywrapstep', ctypes.c_uint16),
-            ('line_length', ctypes.c_uint32),
-            ('mmio_start', ctypes.c_ulong),
-            ('mmio_len', ctypes.c_uint32),
-            ('accel', ctypes.c_uint32),
-            ('reserved', ctypes.c_uint16 * 3),
-        ]
-
-    class VarScreenInfo(ctypes.Structure):
-
-        class FbBitField(ctypes.Structure):
-
-            """The fb_bitfield struct from fb.h."""
+    class Display(object):
+        """
+        Object that represents the EV3 LCD display screen, which implements ANSI escape sequences
+        for cursor positioning, text color, etc.
+        """
+
+        def __init__(self, font="Lat15-TerminusBold24x12"):
+            self.set_font(font)
+            self.reset_screen()
+
+        def set_font(self, font):
+            """
+            Set the LCD console font
+            Parameter:
+                `font` (string): Font name, as found in /usr/share/consolefonts/
+           """
+            self._font = font
+            os.system("setfont %s" % (font))
+            self.reset_screen()
+
+        def set_cursor(self, on=False):
+            """
+            Use ANSI escape sequence to turn LCD screen cursor on or off
+            Parameter:
+                `on` (bool): ``True`` to turn on the screen cursor; default is ``False``
+            """
+            # use escape code to turn cursor on or off
+            print("\x1b[?25%s" % ('h' if on else 'l'), end='')
+
+        def update(self):
+            """
+            NOTE: Not implemented for micropython
+            """
+            pass
+
+        def text_pixels(self, text, clear_screen, x, y, text_color=False, font=None):
+            """
+            NOTE: Not implemented for micropython
+            Draw text on the screen
+
+            Parameters:
+                text (str): The text to draw
```

Comment:
I'd be fine if we dropped all of the parameters definitions in the docstrings for the ones that are not implemented.

Reply:
If we do keep these (not a biggie to me either way) we should make the use of the backtick ` consistent.  This docstring isn't using them for the variables but others are...not all of them though.